<launch>
	<node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true">
		<param name="dev" type="str" value="/dev/input/js0"/>
	</node>
	<!-- Launches the joystick controller -->
	<node name="joystick_controller" pkg="warehouse_interiit" type="joystick_controller.py" output="screen" required="true">
		<!-- Configures the joystick button mapping -->
		<param name="ButtonEmergency"	value="0" />
		<param name="ButtonLand"		value="3" />
		<param name="ButtonTakeoff"		value="2" />

		<!-- Configures the joystick axis mapping -->
		<param name="AxisRoll"			value="0" />
		<param name="AxisPitch"			value="1" />
		<param name="AxisYaw"			value="2" />
		<param name="AxisZ"				value="3" />

		<!-- Configures the joystick mapping -->
		<param name="ScaleRoll"			value="1" />
		<param name="ScalePitch"		value="1" />
		<param name="ScaleYaw"			value="1" />
		<param name="ScaleZ"			value="1" />
	</node>
    <node name="mode_switcher" pkg="warehouse_interiit" type="mode_switcher.py" output="screen" required="true" />
        <!-- joystick mode switcher -->

    <node name="line_detector" pkg="warehouse_interiit" type="line_detector" output="screen" required="true" />
    <node name="strategy" pkg="warehouse_interiit" type="strategy" output="screen" required="true" />
</launch>
